File: frontend/src/shared/api/chat-service.ts
import { apiClient, type HttpClient } from '@/shared/api/client'
import type { ChatCreateDTO, ChatDTO, ChatUpdateDTO } from '@/shared/api/contracts/ai-chat'
import type { RuntimeSessionSummaryDTO } from '@/shared/api/contracts/ai-runtime'

export function createChatService(client: HttpClient = apiClient) {
  return {
    listChats: (): Promise<ChatDTO[]> => client.get('/ai/chats'),
    createChat: (data: ChatCreateDTO): Promise<ChatDTO> => client.post('/ai/chats', data),
    getChat: (chatId: string): Promise<ChatDTO> => client.get(`/ai/chats/${encodeURIComponent(chatId)}`),
    updateChat: (chatId: string, data: ChatUpdateDTO): Promise<ChatDTO> =>
      client.put(`/ai/chats/${encodeURIComponent(chatId)}`, data),
    deleteChat: (chatId: string, expectedVersion: string): Promise<void> =>
      client.delete(`/ai/chats/${encodeURIComponent(chatId)}`, { params: { expectedVersion } }),
    listChatSessions: (chatId: string): Promise<RuntimeSessionSummaryDTO[]> =>
      client.get(`/ai/chats/${encodeURIComponent(chatId)}/sessions`),
  }
}

export const chatService = createChatService()

File: core/src/main/java/fun/fengwk/kkstudio/core/ai/chat/service/ChatService.java
package fun.fengwk.kkstudio.core.ai.chat.service;

import fun.fengwk.kkstudio.share.ai.chat.ChatCreateDTO;
import fun.fengwk.kkstudio.share.ai.chat.ChatDTO;
import fun.fengwk.kkstudio.share.ai.chat.ChatUpdateDTO;

import java.util.List;

/** Chat 集合 CRUD 表面。 */
public interface ChatService {

  List<ChatDTO> listChats();

  ChatDTO getChat(String id);

  ChatDTO createChat(ChatCreateDTO createDTO);

  ChatDTO updateChat(String id, ChatUpdateDTO updateDTO);

  void deleteChat(String id, String expectedVersion);
}
